src/dodal/devices/i22/fswitch.py

```diff
@@ -0,0 +1,43 @@
+import asyncio
+import time
+from enum import Enum
+from typing import Dict
+
+from bluesky.protocols import Reading
+from ophyd_async.core import StandardReadable
+from ophyd_async.core.device import DeviceVector
+from ophyd_async.epics.signal import epics_signal_r
+
+
+class FilterState(str, Enum):
+    IN = "IN"
+    OUT = "OUT"
+
+
+class FSwitch(StandardReadable):
```

Comment:
Are there signals for the desired config fields: lens geometry, cylindrical, lens_material?

Reply:
This is going to be one for the scientists. I imagine it'll be constant (for i22), but I can't see it mentioned in the beamline paper or https://www.diamond.ac.uk/default/Instruments/Soft-Condensed-Matter/small-angle/I22/Ops-modes/I22-TR-SAXS-WAXS.html. I can tag it onto some other questions I've fired at Andy if that's helpful...

Reply:
For i22:

> F-switch only contains one type of lens which are paraboloid, cylindrical, beryllium lenses.

Reply:
Suggest just making these soft signals as per https://github.com/DiamondLightSource/dodal/blob/a3e68efaeae6ff16d74ffd7470f056c978feabd9/src/dodal/devices/focusing_mirror.py#L136, and either pass the initial values into the device constructor or default them to something like `""`.

Reply:
Suggest just making these....

Note: I think we will need to be on the latest `ophyd-async` first. I'm working on it, should be done this morning (https://github.com/DiamondLightSource/dodal/issues/475)

Reply:
Thanks for the suggestion @Tom-Willemsen I think we would like to have some further thoughts on the soft signals/metadata but may add it in a follow up PR.